packages/SwingSet/src/kernel/kernel.js

```diff
@@ -212,17 +296,40 @@ export default function buildKernel(kernelEndowments) {
     notifySubscribersAndQueue(kpid, vatID, subscribers, queue);
   }
 
-  const syscallManager = {
+  // The 'vatPowers' are given to vats as arguments of setup(). They
+  // represent controlled authorities that come from the kernel but that do
+  // not go through the syscall mechanism (so they aren't included in the
+  // replay transcript), so they must not be particularly stateful. These
```

Comment:
Yeah, if any of the `vatPower` methods behave differently from one invocation to the next, then vat code which uses it will not be a deterministic function of the transcript, breaking our orthogonal-persistence model. Those functions could have access to state (they could increment an internal counter, or something), but they must not reveal that to the vat that invokes it.. they have to act as if they were stateless.

That may be too much subtlety to fit into the comment. Maybe I should just drop the "particularly"?

Reply:
I opted to add more text instead